interview-questions.md

```diff
@@ -6,42 +6,71 @@ First, without external resources. Challenge yourself to answer from memory as i
+
+On the other hand, an array's value refers to the actual data stored at a particular index within the array. This can be any JavaScript data type, including numbers, strings, objects, or even other arrays.
 
 3. Compare and contrast arrays and strings. Please include ways in which they are the same and ways in which they are different.
 
-Your answer:
+Your answer: one big difference between arrays and strings are the bracket notation that is used. for strings you use "" and for arrays you use []. other difference would be string and arrays do both use zero indexing. string and arrays also dont take the same data types. arrays can hold numbers, srings. strings really hold more text in quotes. strings and arrays also dont use the same built in methods. 
+
+Researched answer: Similarities:
+Sequential Data Structures: Both arrays and strings are sequential data structures, meaning they store a collection of elements or characters in a specific order.
+
+Zero-based Indexing: Both arrays and strings use zero-based indexing, where the first element or character is accessed using index 0, the second using index 1, and so on.
+
+Length Property: Both arrays and strings have a length property that indicates the number of elements in the array or the number of characters in the string.
+
+Iterable: Both arrays and strings can be iterated over using loops or iteration methods like forEach, for...of, or map.
+
+Differences:
+Mutable vs. Immutable: Arrays are mutable, meaning you can change the elements they contain, add new elements, or remove existing ones. Strings, on the other hand, are immutable, meaning once created, their content cannot be changed. You can create a new string by modifying an existing one, but you cannot directly modify the original string.
+
+Data Type: Arrays can hold elements of any data type, including numbers, strings, objects, or even other arrays. Strings, however, are specifically designed to hold sequences of characters.
+
+Methods and Operations: Arrays have various built-in methods and operations for manipulating their elements, such as push, pop, splice, concat, etc. Strings have methods specific to string manipulation, such as charAt, indexOf, substring, split, slice, etc.
+
+Representation: Arrays are represented by square brackets [ ], while strings are represented by quotes, either single (' ') or double (" ").
 
-Researched answer:
+Memory Allocation: Arrays typically allocate memory for each element separately, while strings might use various techniques like string interning or storage optimizations to manage memory more efficiently.
 
 4. STRETCH: What are the four pillars of object oriented programming?
 
-Researched answer:
+Researched answer: The Four pillars of OOPs, abstraction, encapsulation, inheritance, and polymorphism
```

Comment:
I think it's #3 and #4 that you slacked about your researched answers. I encourage you to get basic defs for the 4 words in #4.  With number three, just think about how you would answer in a verbal interview...would you list off the section type, or just give examples kind of thing.  This is fine today, but next time think about how you'd word it if you had to speak it. Great job!  I hope you learned a lot!